### PR TITLE
feat: make `torch-points3d` builds compatible with Python 3.10

### PR DESCRIPTION
**Pull request by bugra-yilmaz**
We used to supposedly support Python versions `^3.8`, but this wasn't really the case since we used to point to Python 3.8 builds for `torch-cluster`, `torch-scatter` and `torch-sparse` packages, which are publicly available.

In this PR, we update the dependency specifications of `torch-points3d` library to 3.10 builds + point to a newly available, internally stored, `torch-points-kernels` build for Python 3.10.

We don't really need to maintain the older Python version builds since they already exist in our internal PyPI repository.

Making the whole thing more maintainable and the build processes more reproducible is not in the scope of this PR.

Related to https://github.com/promaton/ios/issues/2617.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -30,10 +30,10 @@ omegaconf = "^2.1.1"
 # provided by pyg.org, while torch-points-kernels whell is provided by promaton.
 # Please see OSS README for more details on how to build the wheels.
 
-torch-points-kernels = { version = "==0.7.1+anynp", source = "promaton" }
-torch-cluster = {url = "https://data.pyg.org/whl/torch-1.11.0%2Bcu113/torch_cluster-1.6.0-cp38-cp38-linux_x86_64.whl"}
-torch-scatter = {url = "https://data.pyg.org/whl/torch-1.11.0%2Bcu113/torch_scatter-2.0.9-cp38-cp38-linux_x86_64.whl"}
-torch-sparse = {url = "https://data.pyg.org/whl/torch-1.11.0%2Bcu113/torch_sparse-0.6.13-cp38-cp38-linux_x86_64.whl"}
+torch-points-kernels = { url= "https://pypi.dev.promaton.ai/packages/torch_points_kernels-0.7.1-cp310-cp310-linux_x86_64.whl" }
```
Comment (SeanC-Promaton) [inline pyproject.toml:33]: Just to be sure :thinking: 
Is this one you built + pushed yourself guessing from local from this fork?
https://github.com/promaton/torch-points-kernels

**Review by SeanC-Promaton — APPROVED** [review]
It's fine if you're wanting to go the quick way + just publish some working packages to try out downstream in the product.

I think the part you descoped is important though if we are committing to continued usage :smile:   `Making the whole thing more maintainable and the build processes more reproducible is not in the scope of this PR.`

File: pyproject.toml (hunk 1 above)
Comment (bugra-yilmaz) [reply pyproject.toml:33]: Yes, indeed!

Comment (bugra-yilmaz) [reply pyproject.toml:33]: Actually, `torch-points-kernels` is built and published in a docker container - as explained [here](https://github.com/promaton/ios/tree/main/projects/oss#torch-points-kernels). So, I juggled with the docker stuff to build a version compatible with Python 3.10.

**Comment by bugra-yilmaz** [comment]
@SeanC-Promaton that's a separate KR. For upgrading stuff Python 3.10 I just need one new release for each of the libraries, not really have to revamp the whole build process for each of them. I'll do that later, with the Torch + CUDA upgrades it'll be a bottleneck anyway 😛

**Comment by SeanC-Promaton** [comment]
@bugra-yilmaz just to be sure for this.  Did you run the tests locally manually?  I see the upstream is failing [even](https://github.com/torch-points3d/torch-points3d/actions/runs/9393344674/job/25869233795) for 3.8.

[Edit]: I didn't realize forks didn't run any of the cloned workflows + just saw the zero checks :/

**Pull request merged**